### ISSUE TITLE
Acessing optional serialized fields yields `rest_framework.fields.empty`.

### ISSUE THREAD

**Issue by wkrueger** [oxan/djangorestframework-dataclasses#27]
```py
@dataclass
class SomeClass:
    maybeDog: Optional[Dog]

instance = getInst() # lets say getInst() serializes an input with that dataclass, gets value through validated_data
if instance.maybeDog:  # this is always truthy, because when maybeDog is absent its value is acessed as field.empty
    print('bark')
```

**Comment by oxan** [comment]
See [this note](https://github.com/oxan/djangorestframework-dataclasses/blob/master/CHANGELOG.rst#07-23-october-2020) in the changelog.

**Comment by oxan** [comment]
This issue actually made me think of a way to avoid exposing the `empty` sentinel, which I implemented in f3fff475.
